app/src/main/java/ca/klapstein/baudit/activities/LoginPatientActivity.java

```diff
@@ -20,44 +25,70 @@
 
     private LoginPresenter presenter;
 
+    private EditText usernameInput;
+    private EditText passwordInput;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login_patient);
 
         presenter = new LoginPresenter(this);
-    }
 
-    /**
-     * Attempt a login as a {@code Patient}.
-     *
-     * @return {@code true} if the login was successful, otherwise return {@code false}.
-     */
-    public boolean login() {
-        // TODO: implement
-        return true;
-    }
+        usernameInput = findViewById(R.id.enter_patient_username);
+        passwordInput = findViewById(R.id.enter_patient_password);
 
-    @Override
-    public void setUserNameError() {
+        Button loginButton = findViewById(R.id.login_patient_button);
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.onLoginButtonClicked(
+                    usernameInput.getText().toString(),
+                    passwordInput.getText().toString()
+                );
+            }
+        });
+
+        Button registerButton = findViewById(R.id.register_patient_button);
+        registerButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(getBaseContext(), CreatePatientAccountActivity.class);
+                startActivity(intent);
+                finish();
+            }
+        });
 
+        Button switchLoginButton = findViewById(R.id.log_in_as_care_provider_button);
+        switchLoginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(getBaseContext(), LoginCareProviderActivity.class);
+                startActivity(intent);
+                finish();
+            }
+        });
     }
 
     @Override
-    public void setPasswordError() {
-
+    public void onStart() {
+        super.onStart();
     }
 
     @Override
-    public void setLoginSuccess() {
+    public void onLoginValidationSuccess() {
         Intent intent = new Intent(this, ProblemListActivity.class);
         startActivity(intent);
         finish();
     }
 
     @Override
-    public void onStart() {
-
-        super.onStart();
+    public void onLoginValidationFailure() {
```

Comment:
We should likely highlight the input text boxes with red (and possibly add a sub-text message below the input noting invalid input) to reinforce invalid input. Having only a toast might be a bit confusing. 

e.g. someone logins with invalid input, looks away, misses the toast message and thinks the app login is non responsive.

Obviously this can be implemented later. But, I think it is best practice to highlight inputs that contain invalid values.

Reply:
Makes sense. Changes are now pushed.